client/auth/login.jsx

```diff
@@ -67,16 +67,36 @@ module.exports = React.createClass( {
 
 	componentWillUnmount: function() {
 		AuthStore.off( 'change', this.refreshData );
+		clearTimeout( this.pollTimeout );
+		this.pollTimeout = null;
 	},
 
 	refreshData: function() {
 		this.setState( AuthStore.get() );
 	},
 
+	verifyPushAuthentication: function( pushauth ) {
+		const POLL_INTERVAL = 6000;
+
+		if ( ! this.pollTimeout ) {
+			this.pollTimeout = setTimeout( () => {
+				this.pollTimeout = null;
+				AuthActions.pushAuthLogin( this.state.login, this.state.password, pushauth );
+			}, POLL_INTERVAL );
+		}
+	},
+
 	componentDidUpdate() {
-		if ( this.state.requires2fa && this.state.inProgress === false ) {
+		if ( this.state.requires2fa === 'code' && this.state.inProgress === false ) {
 			ReactDom.findDOMNode( this.refs.auth_code ).focus();
```

Comment:
This PR would be a good opportunity to get rid of the call to `findDOMNode()` and replace it with the newer function-`ref` syntax.

```js
componentDidUpdate() {
	this.authCodeEntry.focus();
}

render() {
	return <input ref={ r => this.authCodeEntry = r } … />;
}
```